src/components/TreeGrid.ts

```diff
@@ -0,0 +1,220 @@
+import _ from 'lodash/fp';
+import { ReactElement, useState } from 'react';
+import { div, h, strong } from 'react-hyperscript-helpers';
+import { Grid } from 'react-virtualized';
+import { Link } from 'src/components/common/Link';
+import { icon } from 'src/components/icons';
+import colors from 'src/libs/colors';
+import { DEFAULT, switchCase, toIndexPairs } from 'src/libs/utils';
+
+export type RowContents = {
+  // The id could be generated by `wrapContent` if we can't rely on the data having an id field.
+  /** The row's unique identifier. */
+  id: number;
+  /** If true, this row has children. */
+  hasChildren: boolean;
+};
+
+export type Column<T extends RowContents> = {
+  /** The column's name, displayed as a column header. */
+  name: string;
+  /** The column's width, in pixels. */
+  width: number;
```

Comment:
Are you sure you want this to be pixels and not a ration (e.g. I would think the columns  would be a flex box).  If I understand this will not adapt to the screen widening

Reply:
The use of fixed widths here is to make the layout easier since Grid uses fixed locations and widths for its elements. But yes, currently you'll get scrollbars if the width or height can't contain the entire tree grid view.

I think this is OK for what it's being used for right now but there's probably a future use use where the column width need to be computed based on the available width instead.

Grid itself is virtualized and can be given a smaller area to render but I don't use that feature in this layout.